### PR TITLE
Fixes deleting exgw pod

### PR DESCRIPTION
**Pull request by trozet**
The ovn_cluster_router policy was being removed for a pod anytime an
exgw was deleted. This is incorrect, because a pod may have more than
one exgw configured. The policy should only be removed for a pod if it
no longer has any exgws configured.

RHBZ: https://bugzilla.redhat.com/show_bug.cgi?id=1917605

Signed-off-by: Tim Rozet <trozet@redhat.com>

### CODE REVIEW

**Comment by trozet** [comment]
@rcarrillocruz @dcbw @nerdalert

File: go-controller/pkg/ovn/egressgw.go [hunk 1]
```diff
@@ -185,6 +181,12 @@ func (oc *Controller) deletePodGWRoutesForNamespace(pod, namespace, node string)
 				// clean up if there are no more routes for this podIP
 				if entry := nsInfo.podExternalRoutes[podIP]; len(entry) == 0 {
 					delete(nsInfo.podExternalRoutes, podIP)
+					// TODO (trozet): use the go bindings here and batch commands
+					// delete the ovn_cluster_router policy if the pod has no more exgws to revert back to normal
+					// default gw behavior
+					if err := oc.delHybridRoutePolicyForPod(net.ParseIP(podIP), node); err != nil {
+						klog.Error(err)
```
Comment (girishmg) [inline go-controller/pkg/ovn/egressgw.go:188]: @trozet the `node` above seems to be the Node on which the External Gateway Pod resided and that pod was deleted from. However, you want that `node` to be the Node on which the podIP resides on, isn't it?

You are walking through each of the Pod in namespace NS and if there are no more external gateways for those pods, then you want to delete the the logical router policie on ovn_cluster_router for that pod.

Comment (trozet) [reply go-controller/pkg/ovn/egressgw.go:188]: great catch @girishmg ! thanks, pushing new version.

**Pull request merged**